products/extensions/b2b-components/index.md

```diff
@@ -20,3 +20,134 @@ In the world of digital B2B commerce, where businesses engage with other compani
+
+## Configuring custom toggles for B2B components
+
+The B2B Components allow merchants to selectively choose and configure B2B capabilities according to their needs. They offer merchants the ability to craft a tailored B2B ecommerce experience for their business partners while also allowing agencies to fine-tune Shopware to meet specific requirements. This means that B2B Components can be individually activated or deactivated for each business partner within the shop.
+
+The following articles will guide you how to do this by creating custom toggles via a plugin for B2B Components (Customer-specific features).
+
+The **Customer-specific features** section on the Customer detail page allows the shop merchant to turn these B2B features on or off for a specific customer. 
+
+![Feature Toggles](../../../assets/b2b-feature-toggles.png)
+
+To achieve this, ACL and address the following cases where functionality may be hidden:
+
+1. If the merchant has not activated a feature for a particular customer, it should be hidden.
+2. If the B2B admin has not granted an employee access to a specific feature, it should not be visible.
+
+Considering these scenarios, we can ensure that the appropriate B2B features are displayed and accessible based on feature toggles and admin-granted permissions.
+
+### Prerequisite
+
+To improve organization and maintain a clear structure, it is advisable to relocate all B2B Components into the `B2B` folder within the Commercial plugin. By doing so, you can centralize the B2B-related functionality, making it easier to locate, manage, and maintain the codebase. This folder structure promotes better separation of concerns and enhances the overall modularity of the application.
+
+```
+├── src
+│   ├── B2B
+│   │   ├── QuickOrder
+│   │   ├── AnotherB2BComponent
+│   │   │   CommercialB2BBundle.php
+...
+
+```
+
+To ensure consistency and clarity, it is recommended to make your B2B Component extend `CommercialB2BBundle` instead of `CommercialBundle` and add the `type` as **B2B** attribute inside the `describeFeatures()` method of each B2B Component. This attribute will help identify and categorize the features specifically related to B2B functionality.
+
+By including `type => 'B2B'` in the `describeFeatures()` method, you can distinguish B2B features from other types of features within your application. This will facilitate easier maintenance, organization, and identification of B2B-related functionalities, ensuring a streamlined development process.
+
+For example, consider the following code snippet:
+
+```php
+namespace Shopware\Commercial\B2B\YourB2BComponent;
+
+class YourB2BComponent extends CommercialB2BBundle
+{
+    public function describeFeatures(): array
+    {
+        return [
+            [
+                ...,
+                'type' => self::TYPE_B2B,
+            ],
+        ];
+    }
+}
+```
+
+## Using feature toggle in Route/API/Controller
+
+### Using new service
```

Comment:
Is this heading necessary to be mentioned ?

Reply:
We can remove it, due the first sentences we said "You can use a new Twig extension called..."